ee/allowedcmd/cmd.go

```diff
@@ -13,17 +13,55 @@ import (
 	"os"
 	"os/exec"
 	"path/filepath"
+
+	"github.com/kolide/launcher/pkg/traces"
 )
 
-type AllowedCommand func(ctx context.Context, arg ...string) (*exec.Cmd, error)
+type AllowedCommand func(ctx context.Context, arg ...string) (*TracedCmd, error)
+
+type TracedCmd struct {
+	*exec.Cmd
+}
+
+// Start overrides the Start method to add tracing before executing the command.
+func (t *TracedCmd) Start() error {
+	_, span := traces.StartSpan(context.Background(), "path", t.Cmd.Path)
```

Comment:
Since `exec.Cmd.ctx` isn't exposed, I think you'd want to add the `ctx` into `TracedCommand` struct so that you can use the ctx here too and make sure that this span attaches to any existing parent spans

Reply:
Maybe the other option would be to update these method signatures to e.g. `Start(ctx context.Context)`? I don't think the embedded ctx is bad here either, though, since it's just following what exec.Cmd also does